app/src/main/java/ch/epfl/sdp/blindly/settings/Settings.kt

```diff
@@ -40,16 +41,30 @@ class Settings : AppCompatActivity() {
         locationText.text = getString(R.string.lausanne_switzerland)
 
         val radiusText = findViewById<TextView>(R.id.radius_text)
-        val radiusSeekBar = findViewById<SeekBar>(R.id.seekBar)
+        val radiusSlider = findViewById<Slider>(R.id.location_slider)
 
-        radiusSeekBar.setOnSeekBarChangeListener(object : SeekBar.OnSeekBarChangeListener {
-            override fun onProgressChanged(seek: SeekBar, progress: Int, fromUser: Boolean) {
-                radiusText.text = getString(R.string.progress_km, progress)
-            }
+        //Update the radius text with initial value, and everytime the slider changes
+        radiusText.text = getString(R.string.progress_km, radiusSlider.value.toInt())
+        radiusSlider.addOnChangeListener { _, value, _ ->
+            radiusText.text = getString(R.string.progress_km, value.toInt())
+        }
 
-            override fun onStartTrackingTouch(seek: SeekBar) {}
-            override fun onStopTrackingTouch(seek: SeekBar) {}
-        })
+        val ageRangeText = findViewById<TextView>(R.id.selected_age_range_text)
+        val ageRangeSlider = findViewById<RangeSlider>(R.id.age_range_slider)
+
+        //Update the selected age range text with initial value, and everytime the slider changes
+        ageRangeText.text = getString(
+            R.string.selected_age_range,
+            ageRangeSlider.values[0].toInt(),
+            ageRangeSlider.values[1].toInt()
+        )
+        ageRangeSlider.addOnChangeListener { _, _, _ ->
+            ageRangeText.text = getString(
+                R.string.selected_age_range,
+                ageRangeSlider.values[0].toInt(),
+                ageRangeSlider.values[1].toInt()
+            )
+        }

```

Comment:
As codeclimate says, I think it might be better to create a private function to contain the lines 56 to 60 that are copy-pasted in lines 62 to 66.

Reply:
Thing is it's already a single function, I only call getString so I thought using a private method was a bit overkill... But if you think it's better then I'll do it